### PR TITLE
snap: use newer json-glib (Fixes: #4099)

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Concept yes please, but CI unhappy: 

    meson.build:1:0: ERROR: Meson version is 0.54.0 but project requires >= 0.55.3

I wonder what we're doing that needs the new json-glib...

**Comment by hughsie** [comment]
> I wonder what we're doing that needs the new json-glib...

I think https://github.com/fwupd/fwupd/pull/4101 "fixes" the bug, but I also think we need this PR too.

**Review by hughsie — DISMISSED** [review]
If CI likes it, so do I. Thanks!

**Comment by superm1** [comment]
> Ohh we need this; the file has usernames and passwords and can't be readable by users.

Hmm, not the default one though in the snap, that's only after it's been modified by the users.  So will need to think about how to solve this considering that.

**Comment by hughsie** [comment]
> Hmm, not the default one though in the snap

Ohh? Maybe we want to do a manual `g_chmod()` after writing the password data? I'm not sure if that's going to make rpm upset, but I think it's okay if it's marked confignoreplace.

**Comment by superm1** [comment]
> > Hmm, not the default one though in the snap
> 
> Ohh? Maybe we want to do a manual `g_chmod()` after writing the password data? I'm not sure if that's going to make rpm upset, but I think it's okay if it's marked confignoreplace.

I think the problem ends up being that someone can modify it by hand.  I suspect doing a fixup in the snap build scripts is better.

**Comment by superm1** [comment]
> > > Hmm, not the default one though in the snap
> > 
> > 
> > Ohh? Maybe we want to do a manual `g_chmod()` after writing the password data? I'm not sure if that's going to make rpm upset, but I think it's okay if it's marked confignoreplace.
> 
> I think the problem ends up being that someone can modify it by hand. I suspect doing a fixup in the snap build scripts is better.

oh but the application programmaticaly sets it.  Let me do the `g_chmod` thing.

File: plugins/redfish/fu-plugin-redfish.c [hunk 1]
```diff
@@ -276,6 +276,9 @@ fu_redfish_plugin_ipmi_create_user(FuPlugin *plugin, GError **error)
 		return FALSE;
 	if (!fu_plugin_set_config_value(plugin, "Password", password_new, error))
 		return FALSE;
+	if (!fu_plugin_set_secure_config_file(plugin, error))
```
Comment (hughsie) [inline plugins/redfish/fu-plugin-redfish.c:279]: Maybe we want a `fu_plugin_set_config_value_secure()` that does this atomically? It seems kinda risky to have a window of time where the file is readable by non-root with a password in.

Comment (superm1) [reply plugins/redfish/fu-plugin-redfish.c:279]: That's true.  Will swap around.

**Review by hughsie — APPROVED** [review]

**Pull request merged**